scripts/biomedical/VirtualMetabolicHuman/format_metabolite.py

```diff
@@ -205,120 +111,156 @@ def add_query_result(df, col, dcid):
                 j += 2
     return df
 
-def main():
-    file_input = sys.argv[1]
-    file_output = sys.argv[2]
-    file_hmdb = sys.argv[3]
-
-    df = pd.read_csv(file_input, sep='\t')
-    hmdbextract(file_hmdb, 'hmdb.csv')
-    dfh = pd.read_csv('hmdb.csv')
-    #inchikey matches with dc 
-    list_inchi = df[['inchiKey']].T.stack().tolist()
-    list_inchi_1 = list_inchi[1:1000]
-    list_inchi_2 = list_inchi[1000:2000]
-    list_inchi_3 = list_inchi[2000:2982]
-    arr_inchi_1 = np.array(list_inchi_1)
-    arr_inchi_2 = np.array(list_inchi_2)
-    arr_inchi_3 = np.array(list_inchi_3)
-
+def inchi_query(arr_inchi):
+    """
+    Queries dc using the python api, to find
+    if the elements of the input list, have a 
+    pre-existing matching inchikey on dc
+    Args:
+        arr_inchi = array with inchikeys to query
+    Returns:
+        result = result of dc query
+    """
     query_str = """
     SELECT DISTINCT ?drug ?id
     WHERE {{
     ?drug typeOf ChemicalCompound .
     ?drug inChIKey ?id .
     ?drug inChIKey {0} .
     }}
-    """.format(arr_inchi_1)
+    """.format(arr_inchi)   
     result = dc.query(query_str)
-    dcid_inch = clean_result(result)
-    df = add_query_result(df, "inchiKey", dcid_inch)
-
+    return result
 
+def hmdb_query(arr_hmdb):
+    """
+    Queries dc using the python api, to find
+    if the elements of the input list, have a 
+    pre-existing matching hmdb ids on dc
+    Args:
+        arr_hmdb = array with hmdb ids to query
+    Returns:
+        result = result of dc query
+    """    
     query_str = """
     SELECT DISTINCT ?drug ?id
     WHERE {{
     ?drug typeOf ChemicalCompound .
-    ?drug inChIKey ?id .
-    ?drug inChIKey {0} .
+    ?drug humanMetabolomeDatabaseID ?id .
+    ?drug humanMetabolomeDatabaseID {0} .
     }}
-    """.format(arr_inchi_2)
-    result = dc.query(query_str)
-    dcid_inch = clean_result(result)
-    df = add_query_result(df, "inchiKey", dcid_inch)
-    #inchi match complete (2)
+    """.format(arr_hmdb)
+    result = dc.query(query_str)    
+    return result
 
+def kegg_query(arr_kegg):
```

Comment:
nice cleanup!

Reply:
I'm noticing now that all the queries have a similar structure. There's more code simplification that can be done -- parameterize the query string to and all the separate _query functions can be collapsed into one.